pyaedt/modeler/Object3d.py

```diff
@@ -1853,18 +1854,91 @@ def remove(self):
         self.padstackmgr.Remove(self.name, True, "", "Project")
 
 
-class CircuitComponent(object):
-    """Manages circuit components.
+class CircuitPins(object):
+    def __init__(self, circuit_comp, pinname):
+        self._circuit_comp = circuit_comp
+        self.name = pinname
+        self.m_Editor = circuit_comp.m_Editor
 
-    Parameters
-    ----------
-    editor : optional
+    @property
+    def location(self):
+        """Pin Position in [x,y] format."""
+        return [_retry_ntimes(30, self.m_Editor.GetComponentPinLocation, self._circuit_comp.composed_name, self.name,
+                              True),
+                _retry_ntimes(30, self.m_Editor.GetComponentPinLocation, self._circuit_comp.composed_name, self.name,
+                              False)]
+
+    @aedt_exception_handler
+    def connect_to_component(self, component_pin):
+        """Connect schematic components.
+
+        Parameters
+        ----------
+        component_pin : :class:`pyaedt.modeler.PrimitivesNexxim.CircuitPins`
+           Component Pin to attach
 
-    units, str, optional
-       The default is ``"mm"``.
-    tabname, str, optional
-       The default is ``"PassedParameterTab"``.
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        """
 
+        page_name = "{}_{}".format(self._circuit_comp.composed_name, self.name)
+        try:
+            x_loc = AEDT_UNITS["Length"][decompose_variable_value(self._circuit_comp.location[0])[1]] * float(
+                decompose_variable_value(self._circuit_comp.location[1])[0])
+        except:
+            x_loc = float(self._circuit_comp.location[0])
+        if self.position[0] < x_loc:
+            angle = 6.28318530717959
+        else:
+            angle = 3.14159265358979
+
+        ret1 = self._circuit_comp._circuit_components.create_page_port(page_name, posx=self.location[0],
+                                                                       posy=self.location[1], angle=angle)
+        try:
+            x_loc = AEDT_UNITS["Length"][decompose_variable_value(component_pin._circuit_comp.location[0])[1]] * float(
+                decompose_variable_value(component_pin._circuit_comp.location[0])[0])
+        except:
+            x_loc = float(self._circuit_comp.location[0])
+        if component_pin.position[0] < x_loc:
+            angle = 6.28318530717959
+        else:
+            angle = 3.14159265358979
```

Comment:
```suggestion
            angle = math.pi
```
Use of math.pi is better like in other cases.